llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -835,7 +835,7 @@ static InstructionsState getSameOpcode(ArrayRef<Value *> VL,
         auto *CallBase = cast<CallInst>(IBase);
         if (Call->getCalledFunction() != CallBase->getCalledFunction())
           return InstructionsState(VL[BaseIndex], nullptr, nullptr);
-        if (Call->hasOperandBundles() &&
+        if (Call->hasOperandBundles() && CallBase->hasOperandBundles() &&
```

Comment:
```suggestion
        if (Call->hasOperandBundles() && (!CallBase->hasOperandBundles() ||
```

Reply:
Is the idea that if CallBase doesn't have an operandBundle is ignored?

Reply:
The idea is that two calls are incompatible if they have different operand bundles. If one has a bundle but another does not, they are still incompatible.

Reply:
Right, but if we have that change we will return true no? i.e `Call->hasOperandBundles() == true` and `!CallBase->hasOperandBundles() == true` will enter the branch. Unless I misunderstood your change

Reply:
Yes, it will execute return. And that's fine, this return returns false, not true

Reply:
Oh ok, We just don't want a fallthrough